src/sql/src/query_model/hir.rs

```diff
@@ -69,6 +73,18 @@ impl FromHir {
                 }
                 box_id
             }
+            HirRelationExpr::Filter { input, predicates } => {
+                let input_box = self.generate_internal(input);
+                // Wrap the input within a select so the referenced columns are
```

Comment:
What is the "expected order" here and why do we need a dedicated box to enforce it?

Reply:
The comment is wrong, sorry. The columns in `input_box` match the order in `input`. However, a `Select` box is needed to install the predicaye.

Reply:
This comment means that we cannot re-use an existing select box, since otherwise we will be installing the filter `before` the projection of the input, which would be wrong. I'll try to make it clearer.